kibble/models/item_index.go

```diff
@@ -83,6 +83,25 @@ func (itemIndex ItemIndex) Set(slug string, item GenericItem) {
 	}
 }
 
+// Set - an item
+func (itemIndex ItemIndex) SetWithStatus(slug string, statusID int, item GenericItem) {
+
+	slugType := getSlugType(slug)
+
+	index, ok := itemIndex[slugType]
+	if !ok {
+		itemIndex[slugType] = make(map[string]GenericItem)
+		index = itemIndex[slugType]
+	}
+
+	// unresolved can be overwritten, empty and item can not
```

Comment:
The grammar on this comment threw me, I don't think we need the comment if we can potentially re-jig the `if`.

Reply:
Whoops that comment is copied over from the original version of this function in `Set()`